### PR TITLE
Feature/fix bitcode warnings

### PR DESCRIPTION
**Pull request by clpolet**
Fix bitcode and nullability warnings + small fix on tracking APIs signature.

### CODE REVIEW

File: Global.xcconfig [hunk 1]
```diff
@@ -8,11 +8,18 @@ MS_SIM_ARCHS = x86_64 i386
 ARCHS = $(MS_ARM_ARCHS)
 VALID_ARCHS = $(MS_ARM_ARCHS)
 
+// This flag adds -fembed-bitcode-marker in the release configuration
```
Comment (jaelim-ms) [inline Global.xcconfig]: If you rename the file, you should update environment variable in Bitrise build.

File: MobileCenterAnalytics/MobileCenterAnalytics/Internals/MSAnalyticsInternal.h [hunk 2]
```diff
@@ -23,7 +25,7 @@
  * @param pageName  page name.
  * @param properties dictionary of properties.
  */
-+ (void)trackPage:(NSString *)pageName withProperties:(NSDictionary *)properties;
++ (void)trackPage:(NSString *)pageName withProperties:(nullable NSDictionary<NSString *, NSString *> *)properties;
```
Comment (jaelim-ms) [inline MobileCenterAnalytics/MobileCenterAnalytics/Internals/MSAnalyticsInternal.h]: What is this change? Merging issue?

Comment (clpolet) [inline MobileCenterAnalytics/MobileCenterAnalytics/Internals/MSAnalyticsInternal.h]: To fix the nullable warnings we had here I added NS_ASSUME_NONNULL_BEGIN/END but doing so I had to explicitly specify nullable parameters.

Comment (jaelim-ms) [inline MobileCenterAnalytics/MobileCenterAnalytics/Internals/MSAnalyticsInternal.h]: Don't mind. I just checked previous PR and found that I missed the change in the header file. 😄 
BTW, I was talking about the key/value types for NSDictionary, not nullable.

File: Global.xcconfig (hunk 1 above)
Comment (jaelim-ms) [inline Global.xcconfig]: In addition to the comment, you might need to change two environment variables.
`VERSION_FILE` and `VERSION_FILENAME`. Eventually we are going to use `VERSION_FILENAME` but just not to break the build, I keep both variables. Please update before you merge to develop.

**Comment by clpolet** [comment]
More unit tests added to improve code coverage.

**Comment by TroubleMakerBen** [comment]
Looking good.

**Pull request merged**